Feat device properties

lekosfmi opened the pull request:
Addressing #27

dkarchmer commented on src/iotile/models/properties.ts:
```diff
@@ -0,0 +1,8 @@
+
+  constructor(datas) {
+    for (let data of datas) {
+      this[data.name] = data.value;
```
I don't like this. Properties should have a real field to store the MAP.
It should then have proper methods to get the data e.g.

```
properties.set('foo', 'bar')
properties.get('foo')
```

This will allow you to add business logic and error checking

lekosfmi replied:
I am making the changes so that `Properties` is an array of `Property` instead.

In the example above, does the `foo` and `bar` represent the `name` and `value` field respectively on the `Property` model?

dkarchmer replied:
Yes, that API is similar to the one you find in many Browser LocalStorage APIs or even the cookie API

dkarchmer replied:
But if you move to using `Property`, then you don't really need get/set functions. The suggestion was only for `Properties`

dkarchmer commented on src/iotile/models/properties.ts:
```diff
@@ -0,0 +1,8 @@
+export class Properties {
```
I am uneasy about having one `Properties` model instead of an array of `Property` records, but other than gut feeling, I don't have a way to justify me worries, so we can leave like this for now

lekosfmi commented:
Changes that were made:
- Changing from `Properties` to `Property`
- Updated device to expect an array of `Property`: `Array<Property>`
- Changing `device.getProperty` and `getDeviceProperties`method to reflect the changes in `Property` model
- Updated testing for `devices` and `property` model respectively

lekosfmi commented:
All of the changes work on `app.iotile.cloud`
